src/verify.integration.test.ts

```diff
@@ -152,3 +152,174 @@ describe("verify(integration)", () => {
+              error: expect.stringMatching("exception"),
+              issued: false
+            }
+          ]
+        },
+        revoked: {
+          revokedOnAny: false,
+          details: [
+            {
+              address: "0x20bc9C354A18C8178A713B9BcCFFaC2152b53990",
+              revoked: false
+            }
+          ]
+        },
+        valid: false
+      })
+    );
+  });
+
+  it("returns true if Mainnet document is valid", async () => {
+    const checkPromises = verifyWithIndividualChecks(documentMainnetValid);
+    const [hash, issued, revoked, valid] = await Promise.all(checkPromises);
+
+    const results = { hash, issued, revoked, valid };
+
+    expect(results).toEqual({
+      hash: { checksumMatch: true },
+      issued: {
+        issuedOnAll: true,
+        details: [
+          {
+            address: "0x007d40224f6562461633ccfbaffd359ebb2fc9ba",
+            issued: true
+          }
+        ]
+      },
+      revoked: {
+        revokedOnAny: false,
+        details: [
+          {
+            address: "0x007d40224f6562461633ccfbaffd359ebb2fc9ba",
+            revoked: false
+          }
+        ]
+      },
+      valid: true
+    });
+  });
+
+  it("returns true if Ropsten document is valid", async () => {
+    const checkPromises = verifyWithIndividualChecks(
+      documentRopstenValid,
+      "ropsten"
+    );
+    const [hash, issued, revoked, valid] = await Promise.all(checkPromises);
+
+    const results = { hash, issued, revoked, valid };
+
+    expect(results).toEqual({
+      hash: { checksumMatch: true },
+      issued: {
+        issuedOnAll: true,
+        details: [
+          {
+            address: "0xc36484efa1544c32ffed2e80a1ea9f0dfc517495",
+            issued: true
+          }
+        ]
+      },
+      revoked: {
+        revokedOnAny: false,
+        details: [
+          {
+            address: "0xc36484efa1544c32ffed2e80a1ea9f0dfc517495",
+            revoked: false
+          }
+        ]
+      },
+      valid: true
+    });
+  });
+
+  it("returns true if Ropsten token is valid", async () => {
+    const checkPromises = verifyWithIndividualChecks(
+      tokenRopstenValid,
+      "ropsten"
+    );
+    const [hash, issued, revoked, valid] = await Promise.all(checkPromises);
+
+    const results = { hash, issued, revoked, valid };
+
+    expect(results).toEqual({
+      hash: { checksumMatch: true },
+      issued: {
+        issuedOnAll: true,
+        details: [
+          {
+            address: "0x48399Fb88bcD031C556F53e93F690EEC07963Af3",
+            issued: true
+          }
+        ]
+      },
+      revoked: {
+        revokedOnAny: false,
+        details: [
+          {
+            address: "0x48399Fb88bcD031C556F53e93F690EEC07963Af3",
+            revoked: false
+          }
+        ]
+      },
+      valid: true
+    });
+  });
+
+  it("returns false if Ropsten token is invalid", async () => {
```

Comment:
Same as above, the test description is not describing enough what we are testing
- document has not been issued
- document has been revoked

What are we testing ?